proxy/cache.go

```diff
@@ -89,6 +91,7 @@ func isCacheable(m *dns.Msg) bool {
 	qName := m.Question[0].Name
 	qType := m.Question[0].Qtype
 
+	//TODO: Do we need to handle cacheMinTTL settings here??
```

Comment:
I think we do, because now we won't add entries with 0 TTL - but that's what user tries to override using `minttl` setting, isn't he?

Reply:
This is actually a bit interesting.

The way things are currently written in this PR, entries that have 0 TTL will *still* not be cached. This is because this TTL check is happening before the entry is packed into the cache (which is where we currently override).

I'm not sure about the use case of 0 TTL DNS responses from upstream servers.
1. Is 0 returned really only in the case of errors or negative responses?
2. Are there legitimate instances where a site wants a user to check their DNS every time they load it?

I'm asking this, because I'm wondering if we should just keep ignoring/not caching 0 TTL responses.

Reply:
I don't see how much different ttl=0 is from ttl=1, for example, in our situation, so why not cache records with ttl=0 if user wants to override minimum TTL?  Otherwise we have to explain to user why we apply min-ttl value to some records, but not apply it to other records.  However, I still think that's what the user wants: if he uses min-ttl value, then he expects the server to cache all records for this amount of time, ignoring the original TTL value.

Maybe @ameshkov has something to add on this?

https://00f.net/2011/11/17/how-long-does-a-dns-ttl-last/
found an interesting article about DNS TTL.